src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpContext.cs

```diff
@@ -332,10 +332,12 @@ private void SetErrorResponseHeaders(int statusCode)
         public unsafe void SendResponseHeaders(bool appCompleted)
         {
             // Verifies we have sent the statuscode before writing a header
-            var reasonPhraseBytes = Encoding.UTF8.GetBytes(ReasonPhrase ?? ReasonPhrases.GetReasonPhrase(StatusCode));
+            var reasonPhrase = string.IsNullOrWhiteSpace(ReasonPhrase) ? ReasonPhrases.GetReasonPhrase(StatusCode) : ReasonPhrase;
```

Comment:
never use OrWhiteSpace is ASP.NET libraries.... it does some strange unicode stuff. Stick with IsNullOrEmpty

Reply:
Was the reason this wasn't returning 500 is because we never sent the reason phrase?

Reply:
It sounds like http_set_response_status_code threw because of a bad reason phrase.

Reply:
It didn't threw exactly but `pInProcessHandler->QueryHttpContext()->GetResponse()->SetStatus(statusCode, pszReason);` call did not succeed.